lib/gui/lvgl/Kconfig.upstream

```diff
@@ -0,0 +1,867 @@
+# Kconfig file for LVGL v8.0
```

Comment:
The Kconfig file for a module usually goes here:
https://github.com/zephyrproject-rtos/zephyr/tree/main/modules

which also helps to ensure the Kconfig menu for that module is only available to users when the module is available.

With this PR, the `LVGL` configuration is a available even when the LVGL module is not present.

Reply:
You mean I should move the entire lvgl configuration from lib/gui/lvgl/ to modules/? Or just the upstream Kconfig?

Reply:
@tejlmand @carlocaione is telling me that actually the entire glue code for LVGL should live in modules/ but I think this is outside the scope of this PR and should be done in a separate one. What do you think?

Reply:
> @tejlmand @carlocaione is telling me that actually the entire glue code for LVGL should live in modules/ but I think this is outside the scope of this PR and should be done in a separate one. What do you think?

I see the mess if we only mode `Kconfig.upstream` into `modules/` and not the rest of the glue code.

If you believe it to be to much to move all glue code with this PR, then please create a followup issue here:
[New Issue](https://github.com/zephyrproject-rtos/zephyr/issues/new?assignees=&labels=Enhancement&template=enhancement.md&title=)

and make a link in this PR, that way this PR can be approved as we then have a followup task.

Reply:
Done: https://github.com/zephyrproject-rtos/zephyr/issues/41628